packages/flutter/lib/src/material/app.dart

```diff
@@ -373,6 +368,7 @@ class _MaterialAppState extends State<MaterialApp> {
         checkerboardOffscreenLayers: widget.checkerboardOffscreenLayers,
         showSemanticsDebugger: widget.showSemanticsDebugger,
         debugShowCheckedModeBanner: widget.debugShowCheckedModeBanner,
+        platform: theme.platform
```

Comment:
this won't animate correctly. Suppose the theme is changed from Android to iOS. AnimatedTheme will switch the platform at t=0.5, but this will switch the platform at t=0.0.